superset/sql_lab.py

```diff
@@ -245,7 +252,22 @@ def execute_sql_statement(
                 query.id,
                 str(query.to_dict()),
             )
-            data = db_engine_spec.fetch_data(cursor, query.limit)
+            # This is a test to see if the query is being
+            # limited by either the dropdown or the sql.
+            # We are testing to see if more rows exist than the limit.
+            data = db_engine_spec.fetch_data(cursor, limited_query)
+            if query.limit is not None and len(data) <= query.limit:
```

Comment:
I think this needs to be:

```suggestion
            if query.limit is None or len(data) <= query.limit:
```

Since if `query.limit is None` the query was also not limited.